src/biogeophys/SurfaceAlbedoType.F90

```diff
@@ -35,6 +35,22 @@ module SurfaceAlbedoType
      real(r8), pointer :: albsoi_col           (:,:) ! col soil albedo: diffuse (col,bnd) [frc]                    
      real(r8), pointer :: albsnd_hst_col       (:,:) ! col snow albedo, direct , for history files (col,bnd) [frc] 
      real(r8), pointer :: albsni_hst_col       (:,:) ! col snow albedo, diffuse, for history files (col,bnd) [frc] 
+! cenlin add new output variables for albedo for history files only
+     real(r8), pointer :: albd_hst_patch       (:,:) ! patch surface albedo (direct) for history files (numrad)                    
```

Comment:
I see that you are following an existing template of adding variables that are meant for history. But why do we do this? Are these variables somehow different and not simply duplicates?

Reply:
The reasons I did this is because those existing template/default albedo variables are assigned to zero when there is no sunlight, which leads to an incorrect daily and monthly mean values as diagnostic output if users choose to output daily or monthly mean. For these new albedo variables, I set their values to NaN when there is no sunlight so that the CLM daily/monthly mean calculations automatically exclude the time points without sunlight.

Reply:
I think I see what you mean, though maybe it's the opposite:
It looks to me as though you're setting _hst variables to spval (which equals 1e36) instead of setting to nan.

Reply:
I prefer to consolidate to one set of variables, so I will try with the old names but with your new way of initializing.

Reply:
Yes, consolidating to one set of variables would be great. You are right about the spval, but then the model history part would skip the spval values when doing the daily and monthly average calculation during output procedure, if I remember correctly.

Reply:
I have looked at this a bit more, and now I see your original point more clearly:
You did not want to change the output of existing variables, so you added new ones that are almost identical, but not exactly. So now I'm leaning in favor of keeping your _hst and _hst2 variables.